suite/tests/client-interface/drx-scattergather.c

```diff
@@ -791,14 +791,20 @@ DECLARE_FUNC_SEH(FUNCNAME(opcode))                        @N@\
         vmovdqu32  zmm1, [REG_XDX]                        @N@\
         movw       dx, 0xffff                             @N@\
         kmovw      k1, edx                                @N@\
+        /* For the first variant below, with the xmm1
+         * index, only the first half of the xmm0 dest
+         * is set by the opcode. So we need to
+         * explicitly zero the remaining part.
+         */                                               @N@\
+        vpxor      xmm0, xmm0, xmm0                       @N@\
         opcode     xmm0 {k1}, [REG_XAX + xmm1 * 4]        @N@\
         vmovdqu32  [REG_XCX], xmm0                        @N@\
         kmovw      k1, edx                                @N@\
         opcode     xmm0 {k1}, [REG_XAX + ymm1 * 4]        @N@\
-        vmovdqu32  [REG_XCX + 16], ymm0                   @N@\
+        vmovdqu32  [REG_XCX + 16], xmm0                   @N@\
```

Comment:
So this is now storing 128 bits instead of the 256 it was storing -- why?  And why is the next offset still 32 bytes further (+48 from +16) as though this were still storing 256 bits?

Reply:
It should always have been storing 128 bit, since that's all that's generated by the instruction above (note that xmm dest).

About the offset: note that it's an offset into a (xmm + ymm + zmm) sized buffer, which is checked for equality after this asm routine is done (see ref_idx64_val32_xmm_ymm_zmm). The three parts correspond to the result of the three executed opcodes (the ones with xmm, ymm and zmm index regs). In this case, we could defrag ref_idx64_val32_xmm_ymm_zmm to remove the zeroes, but I'm choosing uniformity with other tests.

Reply:
Rethinking this... it doesn't hurt to have a stronger check, especially since some vector operations are expected to zero out the larger regs too. Reverted this to ymm and the one below to zmm.